protocol/app/app.go

```diff
@@ -1725,6 +1733,20 @@ func (app *App) PrepareCheckStater(ctx sdk.Context) {
 	if err := app.ModuleManager.PrepareCheckState(ctx); err != nil {
 		panic(err)
 	}
+
+	// Comparing the local orderbook with memclob's orderbook.
+	if app.GrpcStreamingTestClient != nil {
+		app.ClobKeeper.CompareMemclobOrderbookWithLocalOrderbook(
+			ctx,
+			app.GrpcStreamingTestClient.GetOrderbook(0),
+			0,
+		)
+		app.ClobKeeper.CompareMemclobOrderbookWithLocalOrderbook(
+			ctx,
+			app.GrpcStreamingTestClient.GetOrderbook(1),
+			1,
+		)
+	}
```

Comment:
Simple test in `PrepareCheckState` to compare the full node's orderbook against the test client's local orderbook